### ISSUE TITLE
Use API call as data source with Limit and Offset

### ISSUE THREAD

**Issue by yagrasdemonde** [yajra/laravel-datatables#2076]
Hi,

I saw this example : https://datatables.yajrabox.com/collection/github and this issue : https://github.com/yajra/laravel-datatables/issues/1760 because I have a question about another API (Mailjet) which have constraint on results returned.

So I'm trying to be the more simple as possible to explain.

The goal is to show my Mailjet contacts saved into the datatables with service implementation.

Basically, in my ContactsDataTable class, I call API like this : (I'm using a wrapper to do it)
```php
public function dataTable($query)
    {
        return datatables($query);
    }
public function query()
    {
         $mailjet = Mailjet::getClient();
         $filters = [
            // 'Limit' => 1000,//Max Value permitted by API, 10 by default
           //'Offset' => 0 //By default
          ];
         $response = $mailjet->get(Resources::$Contact, [ 'filters' => $filters]);
         return $this->applyScopes($response->getData());
   }
```
The result of $response->getData() is like this
```php
array: [
  0 => [
    "CreatedAt" => "2019-04-12T11:53:26Z"
    "DeliveredCount" => 0
    "Email" => "email1@example.org"
    "ExclusionFromCampaignsUpdatedAt" => "2019-04-28T09:21:35Z"
    "ID" => 1864410583
  ]
  1 => [
    "CreatedAt" => "2019-04-12T12:39:30Z"
    "DeliveredCount" => 0
    "Email" => "email2@example.org"
    "ExclusionFromCampaignsUpdatedAt" => ""
    "ID" => 1864410588
  ]
...
]
```
So, no problem to show them in datatable.

THE Problem :
I have more of 1000 contacts, and so, how can I build my datatable to show and execute with Limit and Offset parameters the others contacts ? 

I can retrieve the total count of all contacts by another request
```php
$filtersCount = [
            'countOnly' => 1
        ];
        $getCountTotal = $mailjet->get(Resources::$Contact, [ 'filters' => $filtersCount]);
        dd($getCountTotal->getCount());
        //return 1367 
```

Is it possible to build the datatable for the first 1000 contacts and when I click on the last pagination button, contacts from 1001 to the end are loading ?
Or what is the good practice to achieve this ?

It will very nice to guide me to work with the Limit parameter of Mailjet API
Thank you 

- PHP Version 7.2
- Laravel Version 5.8.13
- Laravel-Datatables Version 1

**Comment by judgej** [comment]
At the moment, Datatables will accept a collection of ALL the data, and it will cut a slice out of the collection according to the request parameters that it reads. As you have identified, this is not ideal, especially when the data is behind an API with its own paging (offset+length or pagesize).

The hacky workaround for this is to load up your collection with `offset` number of dummy rows then append your external API rows onto the end. That is also not ideal, and will use more resources the deeper into the page count you get.

However, we have developed a very simple fix for this. See PR #2083 for details. It simply involves passing the offset you are using when calling your API to Datatables when supplying your collection.

**Comment by yajra** [comment]
Fixed via #2083.
